src/main/scala/net/jcazevedo/moultingyaml/YamlPrinter.scala

```diff
@@ -0,0 +1,74 @@
+package net.jcazevedo.moultingyaml
+
+import org.yaml.snakeyaml.{Yaml, DumperOptions}
+
+/**
+  * @author Simone D'Avico (simonedavico@gmail.com)
+  *
+  * Created on 22/02/16.
+  */
+trait OptionsMapping[Flow, Scalar] {
+  def corresponding(flow: FlowStyle): Flow
+  def corresponding(scalar: ScalarStyle): Scalar
+}
+
+sealed trait FlowStyle
+case class Auto() extends FlowStyle
+case class Block() extends FlowStyle
+case class Flow() extends FlowStyle
+
+sealed trait ScalarStyle
+object ScalarStyle {
+  def createStyle(char: Char) = CustomScalarStyle(char)
+}
+
+private[moultingyaml] case class CustomScalarStyle(val char: Char) extends ScalarStyle
+case class DoubleQuoted() extends ScalarStyle
+case class SingleQuoted() extends ScalarStyle
+case class Literal() extends ScalarStyle
+case class Plain() extends ScalarStyle
+case class Folded() extends ScalarStyle
+
+abstract class YamlPrinter(flowStyle: FlowStyle,
+                           scalarStyle: ScalarStyle) extends (YamlValue => String) {
+  def apply(value: YamlValue): String
+}
+
+trait SnakeYAMLOptionsMapping extends OptionsMapping[DumperOptions.FlowStyle, DumperOptions.ScalarStyle] {
```

Comment:
I thought about a similar solution as well, but it seemed to me I was coupling too much my types to SnakeYAML's `DumperOption`s. 

So, something like

```scala
sealed trait ScalarStyle { def toDumperOption: DumperOptions.ScalarStyle }
case object Literal extends ScalarStyle {  
    def toDumperOption = DumperOptions.ScalarStyle.LITERAL
}
```

would be closer to what you were thinking?